pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -107,7 +107,7 @@ func BuildSidecarVirtualHostsFromConfigAndRegistry(
 	for fqdn := range missing {
 		svc := serviceRegistry[fqdn]
 		for _, port := range svc.Ports {
-			if port.Protocol.IsHTTP() {
+			if port.Protocol.IsHTTP() || (util.IsProtocolSniffingEnabled(node) && port.Protocol.IsUnsupported()) {
```

Comment:
btw, there is another bug in RDS that you need to fix..  When you install a HCM in a auto listener that is bound to x.x.x.x:port (where x.x.x.x != 0.0.0.0), then the RDS response from Pilot will return all possible hosts from that port. Not just the host for the specific service. Keep note to fix this in the subsequent PR please.

Reply:
Ok. I will add a todo here.